api/ocp/scc.go

```diff
@@ -0,0 +1,37 @@
+package ocp
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"text/template"
+
+	secv1 "github.com/openshift/api/security/v1"
+	"sigs.k8s.io/yaml"
+)
+
+//go:embed scc.yaml
+var sccYAMLTemplate string
+
+// NewSecurityContextConstraints loads the embedded SCC YAML template, replaces the namespace,
+// and returns it as a SecurityContextConstraints object
+func NewSecurityContextConstraints(name string, namespace string) (*secv1.SecurityContextConstraints, error) {
+	tmpl, err := template.New("scc").Parse(sccYAMLTemplate)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing template: %v", err)
+	}
+
+	var buf bytes.Buffer
+	err = tmpl.Execute(&buf, map[string]string{"Name": name, "Namespace": namespace})
+	if err != nil {
+		return nil, fmt.Errorf("error executing template: %v", err)
+	}
+
+	scc := &secv1.SecurityContextConstraints{}
+	err = yaml.UnmarshalStrict(buf.Bytes(), scc)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshaling YAML: %v", err)
+	}
+
+	return scc, nil
+}
```

Comment:
Why do we need this? where is it used? 
The assumption is that a template is enough for users. 
They will either add it to their own automation or use it as part of a halm chart

Reply:
the ocs-operator imports all the SCCs and uses this function, to apply it during initialization

ref: https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/ocsinitialization/sccs.go